CMakeLists.txt

```diff
@@ -863,3 +864,19 @@ IF ("${VERSION_PRERELEASE}" STREQUAL "-DEV" AND EXISTS "${CMAKE_SOURCE_DIR}/.git
     ENDIF (LIBUV_FOUND)
 ENDIF ()
 
+
+OPTION(WITH_H2OLOG "use h2olog(1)" OFF)
+if (WITH_H2OLOG)
+    IF (NOT CMAKE_SYSTEM_NAME STREQUAL "Linux")
+        MESSAGE(FATAL "h2olog supports only Linux")
+    ENDIF ()
+
+    IF (NOT WITH_DTRACE)
+        MESSAGE(FATAL "h2olog requires DTrace support")
+    ENDIF ()
+
+    ADD_SUBDIRECTORY(misc/h2olog)
```

Comment:
I think h2olog can have its own directory (i.e. `./h2olog` rather than `misc/h2olog)`.

Also, I am not sure if we need a separate CMakeLists.txt. To me, it seems that there are certain complexities around having a separate file (like use of `gitrev`).

Reply:
> Also, I am not sure if we need a separate CMakeLists.txt. To me, it seems that there are certain complexities around having a separate file (like use of gitrev).

I'd like to have a separate CMakeLists.txt even if it introduces a bit of complexity.

This is because `h2olog` is literally a "project" of CMake: a separate set of source files and dependencies. `h2o/CMekeLists.txt` is already large (but of course it's necessary!), so I don't want to add lines to `h2o/CMakeLists.txt`, which are irrelevant to how to build h2o.

Reply:
I'm not sure if I agree, especially if we think that h2olog should be built if the dependencies are met (see https://github.com/h2o/h2o/pull/2353/files#r439983898).

Reply:
Discussed. Since https://github.com/h2o/h2o/pull/2353#discussion_r439983898 needs to detect h2olog dependencies in `/CMakeLists.txt`, it's simpler to merge `h2olog/CMakeLists.txt` to the root's. It makes sense.